Replicator/IncomingRev+Blobs.cc

```diff
@@ -30,12 +30,33 @@ namespace litecore { namespace repl {
 
     // Looks for another blob to download; when they're all done, finishes up the revision.
     void IncomingRev::fetchNextBlob() {
-        while (_blob != _pendingBlobs.end()) {
+        // Invariant: _danglingBlobBegin == _pendingBlobs.begin() + n, for some n where 0 <= n && n <= _pendingBlobs.size()
```

Comment:
If this is an invariant, why is it not an assertion?

Reply:
I sometimes put simple assertion or invariant in the comment to help understand the code and easily verified, to avoid overcrowd the runtime assertion.